### PR TITLE
Fix/region data model

### PR DESCRIPTION
**Pull request by VivianCheungFW**

### CODE REVIEW

File: migrations/etl/1_Mine_profile_detail_location.sql [hunk 1]
```diff
@@ -108,7 +108,7 @@ BEGIN
         create_timestamp    ,
         update_user         ,
         update_timestamp    ,
-        major               )
+        major_mine_ind               )
```
Comment (NWCalvank) [inline migrations/etl/1_Mine_profile_detail_location.sql]: Thanks for fixing ETL for me! Should probably line up the closing parenthesis with the commas, though

File: migrations/sql/V1.5__Alter_Mine_Region.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    ADD effective_date date NOT NULL DEFAULT now(),
+    ADD expiry_date date NOT NULL DEFAULT '9999-12-31'::date
+;
+
+ALTER TABLE mine_region_code RENAME region_code TO mine_region_code;
+
+ALTER TABLE mine_detail 
+    ADD mine_region character varying(2) NOT NULL,
+    ADD FOREIGN KEY(mine_region) REFERENCES mine_region_code(mine_region_code) 
+;
```
Comment (FreshGyan) [inline migrations/sql/V1.5__Alter_Mine_Region.sql]: Can you add the corresponding Undo SQL script for this?

File: python-backend/app/api/mines/mine/models/mine.py [hunk 3]
```diff
@@ -97,17 +96,22 @@ class MineDetail(AuditMixin, Base):
...
-    major_mine_ind = db.Column(db.Boolean, nullable=False, default=False)
+    major_mine_ind = db.Column(db.Boolean, nullable=False, default=False) 
+    mine_region = db.Column(db.String(2), db.ForeignKey('mine_region_code.mine_region_code'))
+
 
     def __repr__(self):
         return '<MineDetail %r>' % self.mine_guid
 
     def json(self):
+        region_description=MineRegionCode.find_by_region_code(self.mine_region).json()
```
Comment (NWCalvank) [inline python-backend/app/api/mines/mine/models/mine.py]: I'm not familiar with this part of the codebase, but is there a need to convert it to JSON before pulling out the description on line 114?

File: python-backend/app/api/mines/mine/resources/mine.py [hunk 4]
```diff
@@ -122,6 +120,7 @@ def post(self, mine_no_or_guid=None):
                 mine_name=data['name'],
                 mine_note=note if note else '',
                 major_mine_ind=major_mine_ind,
+                mine_region=region,
```
Comment (NWCalvank) [inline python-backend/app/api/mines/mine/resources/mine.py]: For consistency, should we rename the `region` key to `mine_region`? Like how `major` was renamed to `major_mine_ind`

File: python-backend/app/api/mines/region/models/region.py [hunk 5]
```diff
@@ -6,47 +6,28 @@
...
-        }
-
-    @classmethod
-    def find_by_mine_region_guid(cls, _id):
-        return cls.query.filter_by(mine_region_guid=_id).first()
+        return cls.query.filter_by(mine_region_code=_code).first()
  
- 
\ No newline at end of file
+
```
Comment (NWCalvank) [inline python-backend/app/api/mines/region/models/region.py]: Remove trailing white space

File: python-backend/app/api/mines/mine/models/mine.py (hunk 3 above)
Comment (FreshGyan) [inline python-backend/app/api/mines/mine/models/mine.py]: Optimization bottleneck

**Pull request closed without merging**